This is a portion of cheribsd.patch, extracted into a separate patch as we only want
to apply it to target builds. Rather than using drm_uptr_t (as the origional patch did)
we use __u64ptr to match the kernel header files.

Upstream-Status: Inappropriate [CHERI specific]

Index: include/drm-uapi/amdgpu_drm.h
===================================================================
--- include/drm-uapi/amdgpu_drm.h.orig
+++ include/drm-uapi/amdgpu_drm.h
@@ -211,7 +211,7 @@ struct drm_amdgpu_bo_list_in {
 	/** Size of each element describing BO */
 	__u32 bo_info_size;
 	/** Pointer to array describing BOs */
-	__u64 bo_info_ptr;
+	__u64ptr bo_info_ptr;
 };
 
 struct drm_amdgpu_bo_list_entry {
@@ -365,7 +365,7 @@ union drm_amdgpu_sched {
 #define AMDGPU_GEM_USERPTR_REGISTER	(1 << 3)
 
 struct drm_amdgpu_gem_userptr {
-	__u64		addr;
+	__u64ptr 	addr;
 	__u64		size;
 	/* AMDGPU_GEM_USERPTR_* */
 	__u32		flags;
@@ -502,7 +502,7 @@ struct drm_amdgpu_fence {
 
 struct drm_amdgpu_wait_fences_in {
 	/** This points to uint64_t * which points to fences */
-	__u64 fences;
+	__u64ptr fences;
 	__u32 fence_count;
 	__u32 wait_all;
 	__u64 timeout_ns;
@@ -528,7 +528,7 @@ struct drm_amdgpu_gem_op {
 	/** AMDGPU_GEM_OP_* */
 	__u32	op;
 	/** Input or return value */
-	__u64	value;
+	__u64ptr	value;
 };
 
 #define AMDGPU_VA_OP_MAP			1
@@ -613,7 +613,7 @@ struct drm_amdgpu_gem_va {
 struct drm_amdgpu_cs_chunk {
 	__u32		chunk_id;
 	__u32		length_dw;
-	__u64		chunk_data;
+	__u64ptr	chunk_data;
 };
 
 struct drm_amdgpu_cs_in {
@@ -624,7 +624,7 @@ struct drm_amdgpu_cs_in {
 	__u32		num_chunks;
 	__u32		flags;
 	/** this points to __u64 * which point to cs chunks */
-	__u64		chunks;
+	__u64ptr	chunks;
 };
 
 struct drm_amdgpu_cs_out {
@@ -933,7 +933,7 @@ struct drm_amdgpu_query_fw {
 /* Input structure for the INFO ioctl */
 struct drm_amdgpu_info {
 	/* Where the return value will be stored */
-	__u64 return_pointer;
+	__u64ptr return_pointer;
 	/* The size of the return value. Just like "size" in "snprintf",
 	 * it limits how many bytes the kernel can write. */
 	__u32 return_size;
Index: include/drm-uapi/drm.h
===================================================================
--- include/drm-uapi/drm.h.orig
+++ include/drm-uapi/drm.h
@@ -922,7 +932,7 @@ struct drm_syncobj_transfer {
 #define DRM_SYNCOBJ_WAIT_FLAGS_WAIT_AVAILABLE (1 << 2) /* wait for time point to become available */
 #define DRM_SYNCOBJ_WAIT_FLAGS_WAIT_DEADLINE (1 << 3) /* set fence deadline to deadline_nsec */
 struct drm_syncobj_wait {
-	__u64 handles;
+	__u64ptr handles;
 	/* absolute timeout */
 	__s64 timeout_nsec;
 	__u32 count_handles;
@@ -940,9 +950,9 @@ struct drm_syncobj_wait {
 };
 
 struct drm_syncobj_timeline_wait {
-	__u64 handles;
+	__u64ptr handles;
 	/* wait on specific timeline point for every handles*/
-	__u64 points;
+	__u64ptr points;
 	/* absolute timeout */
 	__s64 timeout_nsec;
 	__u32 count_handles;
@@ -982,15 +992,15 @@ struct drm_syncobj_eventfd {
 
 
 struct drm_syncobj_array {
-	__u64 handles;
+	__u64ptr handles;
 	__u32 count_handles;
 	__u32 pad;
 };
 
 #define DRM_SYNCOBJ_QUERY_FLAGS_LAST_SUBMITTED (1 << 0) /* last available point on timeline syncobj */
 struct drm_syncobj_timeline_array {
-	__u64 handles;
-	__u64 points;
+	__u64ptr handles;
+	__u64ptr points;
 	__u32 count_handles;
 	__u32 flags;
 };
@@ -1015,7 +1025,7 @@ struct drm_crtc_queue_sequence {
 	__u32 crtc_id;
 	__u32 flags;
 	__u64 sequence;		/* on input, target sequence. on output, actual sequence */
-	__u64 user_data;	/* user data passed to event */
+	__u64ptr user_data;	/* user data passed to event */
 };
 
 #if defined(__cplusplus)
@@ -1341,7 +1351,7 @@ struct drm_event {
 
 struct drm_event_vblank {
 	struct drm_event base;
-	__u64 user_data;
+	__u64ptr user_data;
 	__u32 tv_sec;
 	__u32 tv_usec;
 	__u32 sequence;
@@ -1353,7 +1363,7 @@ struct drm_event_vblank {
  */
 struct drm_event_crtc_sequence {
 	struct drm_event	base;
-	__u64			user_data;
+	__u64ptr			user_data;
 	__s64			time_ns;
 	__u64			sequence;
 };
Index: include/drm-uapi/drm_mode.h
===================================================================
--- include/drm-uapi/drm_mode.h.orig
+++ include/drm-uapi/drm_mode.h
@@ -260,10 +260,10 @@ struct drm_mode_modeinfo {
 };
 
 struct drm_mode_card_res {
-	__u64 fb_id_ptr;
-	__u64 crtc_id_ptr;
-	__u64 connector_id_ptr;
-	__u64 encoder_id_ptr;
+	__u64ptr fb_id_ptr;
+	__u64ptr crtc_id_ptr;
+	__u64ptr connector_id_ptr;
+	__u64ptr encoder_id_ptr;
 	__u32 count_fbs;
 	__u32 count_crtcs;
 	__u32 count_connectors;
@@ -275,7 +275,7 @@ struct drm_mode_card_res {
 };
 
 struct drm_mode_crtc {
-	__u64 set_connectors_ptr;
+	__u64ptr set_connectors_ptr;
 	__u32 count_connectors;
 
 	__u32 crtc_id; /**< Id */
@@ -354,11 +354,11 @@ struct drm_mode_get_plane {
 	 * @format_type_ptr: Pointer to ``__u32`` array of formats that are
 	 * supported by the plane. These formats do not require modifiers.
 	 */
-	__u64 format_type_ptr;
+	__u64ptr format_type_ptr;
 };
 
 struct drm_mode_get_plane_res {
-	__u64 plane_id_ptr;
+	__u64ptr plane_id_ptr;
 	__u32 count_planes;
 };
 
@@ -457,13 +457,13 @@ enum drm_mode_subconnector {
  */
 struct drm_mode_get_connector {
 	/** @encoders_ptr: Pointer to ``__u32`` array of object IDs. */
-	__u64 encoders_ptr;
+	__u64ptr encoders_ptr;
 	/** @modes_ptr: Pointer to struct drm_mode_modeinfo array. */
-	__u64 modes_ptr;
+	__u64ptr modes_ptr;
 	/** @props_ptr: Pointer to ``__u32`` array of property IDs. */
-	__u64 props_ptr;
+	__u64ptr props_ptr;
 	/** @prop_values_ptr: Pointer to ``__u64`` array of property values. */
-	__u64 prop_values_ptr;
+	__u64ptr prop_values_ptr;
 
 	/** @count_modes: Number of modes. */
 	__u32 count_modes;
@@ -589,9 +589,9 @@ struct drm_mode_property_enum {
  */
 struct drm_mode_get_property {
 	/** @values_ptr: Pointer to a ``__u64`` array. */
-	__u64 values_ptr;
+	__u64ptr values_ptr;
 	/** @enum_blob_ptr: Pointer to a struct drm_mode_property_enum array. */
-	__u64 enum_blob_ptr;
+	__u64ptr enum_blob_ptr;
 
 	/**
 	 * @prop_id: Object ID of the property which should be retrieved. Set
@@ -632,8 +632,8 @@ struct drm_mode_connector_set_property {
 #define DRM_MODE_OBJECT_ANY 0
 
 struct drm_mode_obj_get_properties {
-	__u64 props_ptr;
-	__u64 prop_values_ptr;
+	__u64ptr props_ptr;
+	__u64ptr prop_values_ptr;
 	__u32 count_props;
 	__u32 obj_id;
 	__u32 obj_type;
@@ -649,7 +649,7 @@ struct drm_mode_obj_set_property {
 struct drm_mode_get_blob {
 	__u32 blob_id;
 	__u32 length;
-	__u64 data;
+	__u64ptr data;
 };
 
 struct drm_mode_fb_cmd {
@@ -773,7 +773,7 @@ struct drm_mode_fb_dirty_cmd {
 	__u32 flags;
 	__u32 color;
 	__u32 num_clips;
-	__u64 clips_ptr;
+	__u64ptr clips_ptr;
 };
 
 struct drm_mode_mode_cmd {
@@ -828,9 +828,9 @@ struct drm_mode_crtc_lut {
 	__u32 gamma_size;
 
 	/* pointers to arrays */
-	__u64 red;
-	__u64 green;
-	__u64 blue;
+	__u64ptr red;
+	__u64ptr green;
+	__u64ptr blue;
 };
 
 struct drm_color_ctm {
@@ -1018,7 +1018,7 @@ struct drm_mode_crtc_page_flip {
 	__u32 fb_id;
 	__u32 flags;
 	__u32 reserved;
-	__u64 user_data;
+	__u64ptr user_data;
 };
 
 /*
@@ -1046,7 +1046,7 @@ struct drm_mode_crtc_page_flip_target {
 	__u32 fb_id;
 	__u32 flags;
 	__u32 sequence;
-	__u64 user_data;
+	__u64ptr user_data;
 };
 
 /**
@@ -1147,12 +1147,12 @@ struct drm_mode_destroy_dumb {
 struct drm_mode_atomic {
 	__u32 flags;
 	__u32 count_objs;
-	__u64 objs_ptr;
-	__u64 count_props_ptr;
-	__u64 props_ptr;
-	__u64 prop_values_ptr;
-	__u64 reserved;
-	__u64 user_data;
+	__u64ptr objs_ptr;
+	__u64ptr count_props_ptr;
+	__u64ptr props_ptr;
+	__u64ptr prop_values_ptr;
+	__u64ptr reserved;
+	__u64ptr user_data;
 };
 
 struct drm_format_modifier_blob {
@@ -1213,7 +1213,7 @@ struct drm_format_modifier {
  */
 struct drm_mode_create_blob {
 	/** @data: Pointer to data to copy. */
-	__u64 data;
+	__u64ptr data;
 	/** @length: Length of data to copy. */
 	__u32 length;
 	/** @blob_id: Return: new property ID. */
@@ -1247,7 +1247,7 @@ struct drm_mode_destroy_blob {
  */
 struct drm_mode_create_lease {
 	/** @object_ids: Pointer to array of object ids (__u32) */
-	__u64 object_ids;
+	__u64ptr object_ids;
 	/** @object_count: Number of object ids */
 	__u32 object_count;
 	/** @flags: flags for new FD (O_CLOEXEC, etc) */
@@ -1283,7 +1283,7 @@ struct drm_mode_list_lessees {
 	 *
 	 * Pointer to __u64 array of lessee ids
 	 */
-	__u64 lessees_ptr;
+	__u64ptr lessees_ptr;
 };
 
 /**
@@ -1310,7 +1310,7 @@ struct drm_mode_get_lease {
 	 *
 	 * Pointer to __u32 array of object ids.
 	 */
-	__u64 objects_ptr;
+	__u64ptr objects_ptr;
 };
 
 /**
Index: include/drm-uapi/i915_drm.h
===================================================================
--- include/drm-uapi/i915_drm.h.orig
+++ include/drm-uapi/i915_drm.h
@@ -2651,7 +2651,7 @@ struct drm_i915_gem_userptr {
 	 *
 	 * Needs to be aligned to PAGE_SIZE.
 	 */
-	__u64 user_ptr;
+	__u64ptr user_ptr;
 
 	/**
 	 * @user_size:
Index: include/drm-uapi/panfrost_drm.h
===================================================================
--- include/drm-uapi/panfrost_drm.h.orig
+++ include/drm-uapi/panfrost_drm.h
@@ -52,7 +52,7 @@ struct drm_panfrost_submit {
 	__u64 jc;
 
 	/** An optional array of sync objects to wait on before starting this job. */
-	__u64 in_syncs;
+	uintptr_t in_syncs;
 
 	/** Number of sync objects to wait on before starting this job. */
 	__u32 in_sync_count;
@@ -61,7 +61,7 @@ struct drm_panfrost_submit {
 	__u32 out_sync;
 
 	/** Pointer to a u32 array of the BOs that are referenced by the job. */
-	__u64 bo_handles;
+	uintptr_t bo_handles;
 
 	/** Number of BO handles passed in (size is that times 4). */
 	__u32 bo_handle_count;
